src/libraries/Common/src/Interop/Unix/System.Native/Interop.Aio.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct IoEvent
```

Comment:
If you add structs that exactly match native structs, it would be nice to validate this with a shim struct and some asserts.
like is done here: https://github.com/dotnet/runtime/blob/6e83eafe49ce3c55e9f68ae669f24957e5b482f8/src/libraries/Native/Unix/System.Native/pal_networking.h#L290-L296, https://github.com/dotnet/runtime/blob/6e83eafe49ce3c55e9f68ae669f24957e5b482f8/src/libraries/Native/Unix/System.Native/pal_networking.c#L160-L165